examples/cpp/benchmark/README.md

```diff
@@ -0,0 +1,287 @@
+* [Run the example](#run-the-example)
+* [Configuration](#configuration)
+* [XML profile playground](#xml-profile-playground)
+
+## Description of the example
+
+Each example application (publisher and subscriber) creates the required DDS entities per case.
+Both publisher and subscriber inherit from the corresponding listener class, overriding the listener's method associated to each event.
+When an event occurs, the callback method is triggered.
+
+If the environment does not specify the expected configuration, they take the default configuration per entity.
+
+Moreover, this example extends the configuration options and message type size to showcase the results on the amount of messages transported during each sample time of the whole example duration for these conditions.
+
+## Run the example
+
+To launch this example, two different terminals are required.
+One of them will run the publisher example application, and the other will run the subscriber application.
+
+### Benchmark publisher
+
+* Ubuntu ( / MacOS )
+
+    ```shell
+    user@machine:example_path$ ./benchmark publisher
+    Publisher running for 10000 milliseconds. Please press Ctrl+C to stop the Publisher at any time.
+    ```
+
+* Windows
+
+    ```powershell
+    example_path> benchmark.exe publisher
+    Publisher running for 10000 milliseconds. Please press Ctrl+C to stop the Publisher at any time.
+    ```
+
+### Benchmark subscriber
+
+* Ubuntu ( / MacOS )
+
+    ```shell
+    user@machine:example_path$ ./benchmark subscriber
+    Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+    ```
+
+* Windows
+
+    ```powershell
+    example_path> benchmark.exe subscriber
+    Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+    ```
+
+All the example available flags can be queried running the executable with the ``-h`` or ``--help`` flag.
+
+### Expected output
+
+Regardless of which application is run first, since the publisher will not start sending data until a subscriber is discovered, the expected output both for publishers and subscribers is a first displayed message acknowledging the match, followed by the sent and received data in each application, and finishing showing in the publisher the amount of data shared in each sample time during the running time.
+
+### Benchmark publisher
+
+```shell
+Publisher running for 1000 milliseconds. Please press Ctrl+C to stop the Publisher at any time.
+Subscriber matched.
+Publisher matched.
+First Sample with index: '0' (8388608 Bytes) SENT
+Sample with index: '1' (8388608 Bytes) RECEIVED
+Sample with index: '0' (8388608 Bytes) SENT
+Sample with index: '1' (8388608 Bytes) RECEIVED
+Sample with index: '0' (8388608 Bytes) SENT
+Sample with index: '1' (8388608 Bytes) RECEIVED
+Sample with index: '2' (8388608 Bytes) SENT
+Sample with index: '3' (8388608 Bytes) RECEIVED
+Sample with index: '4' (8388608 Bytes) SENT
+Sample with index: '5' (8388608 Bytes) RECEIVED
```

Comment:
Testing locally I found a couple of strange things to me in the publisher side:
* The first index seems to be sent and received twice
* The RECEIVED is only printed for odd sample idexes (1,3,5,...)
Is that behavior correct ?

Reply:
The problem of repetting the first index is fix.
The RECEIVED is only printed for odd because they are sent by the subscriber and receive in the publisher, and the evens are sent by the publisher and receive by the subscriber

Reply:
Yeah, I reviewed the logic and I first thought that the subscriber was not incrementing the index but it is doing it, so it is expected.

Reply:
Please, update the commands output since the `index 0, index 1, index 0,...` behavior was already corrected and no longer exists